Frame.cc

```diff
@@ -484,14 +482,14 @@ void Frame::setChannelCache(size_t channel, size_t stokes) {
     }
 }
 
-void Frame::getChannelMatrix(casacore::Matrix<float>& chanMatrix, size_t channel, size_t stokes) {
+void Frame::getChannelMatrix(std::vector<float>& chanMatrix, size_t channel, size_t stokes) {
     // fill matrix for given channel and stokes
     casacore::Slicer section = getChannelMatrixSlicer(channel, stokes);
     casacore::Array<float> tmp;
     // slice image data
     std::unique_lock<std::mutex> guard(latticeMutex);
     loader->loadData(FileInfo::Data::XYZW).getSlice(tmp, section, true);
-    chanMatrix.reference(tmp);
+    chanMatrix = tmp.tovector();
```

Comment:
This causes a new vector to be allocated in `.tovector`, which is then copied to `chanMatrix`. If you change this to `tmp.tovector(chanMatrix)`, that will reuse `chanMatrix` directly and avoid the copy.

Reply:
Done.